.github/workflows/test.yml

```diff
@@ -14,34 +14,34 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [macos-latest, ubuntu-18.04, ubuntu-20.04, ubuntu-22.04, windows-latest]
+        os: [macos-latest, ubuntu-20.04, ubuntu-22.04, windows-latest]
         python-version: ['3.7', '3.8', '3.9', '3.10']
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
 
       - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python-version }}
 
-      - uses: actions/cache@v1
+      - uses: actions/cache@v3
```

Comment:
This gets rid of a ton of deprecation warnings when viewing Action results, e[.g. as seen in this run](https://github.com/sscpac/statick/actions/runs/3563665572).   I went ahead and updated all versions that weren't throwing warnings just as general maintenance.